src/Microsoft.OData.Cli/GenerateCommand.cs

```
@@ -201,71 +221,162 @@ private async Task<int> HandleGenerateCommand(GenerateOptions options, IConsole
             return 0;
         }
 
-        private ServiceConfiguration GetServiceConfiguration(GenerateOptions generateOptions)
+        /// <summary>
+        /// Get a <see cref="ServiceConfiguration"/> object from <paramref name="generateOptions"/>
+        /// </summary>
+        /// <typeparam name="TServiceConfig">Type of <see cref="ServiceConfiguration"/> to return</typeparam>
+        /// <param name="generateOptions">Source options used to populate the service configuration</param>
+        /// <returns>New <typeparamref name="TServiceConfig"/> from <paramref name="generateOptions"/></returns>
+        private TServiceConfig GetServiceConfiguration<TServiceConfig>(GenerateOptions generateOptions)
+            where TServiceConfig : ServiceConfiguration, new()
+        {
+            TServiceConfig serviceConfig = null;
+
+            if (generateOptions != null)
```

Consider returning immediately after failed pre-conditions for better readability:

```c#
if (generateOptions == null)
{
   // return or throw exception
}

...
```

I think we should throw an exception instead of returning `null` if the `generateOptions` is not provided.

---

Updated method to throw an exception on null argument